### PR TITLE
Fix/correct creation hints

### PR DESCRIPTION
**Pull request by stnmtz**

### CODE REVIEW

**Review by jkoenig134 — APPROVED** [review]

**Comment by jkoenig134** [comment]
@slavistan you used the interface in the class which completely disabled all validation.

File: src/attributes/IQLQuery.ts [hunk 1]
```diff
@@ -17,15 +17,10 @@ export interface IIQLQueryCreationHints extends ISerializable {
...
-        customValidator: (v) =>
-            !AttributeValues.Identity.TYPE_NAMES.includes(v)
-                ? `must be one of: ${AttributeValues.Identity.TYPE_NAMES_STRINGIFIED}`
-                : undefined
-    })
+    @validate({ customValidator: IdentityAttribute.validateTypeName })
     public valueType: AttributeValues.Identity.TypeName
 
-    @serialize()
+    @serialize({ type: String })
```
Comment (slavistan) [inline src/attributes/IQLQuery.ts:23]: What's the reason we have to explicitly provide a `@serialize` decorator here? I was under the impression that built-in types are handled automatically (given an empty `@serialize()`). Is this due to the fact that we're using an array?

Comment (jkoenig134) [reply src/attributes/IQLQuery.ts:23]: The decorator was there before, we need that for every field.

The added type is necessary b/c the reflection doesn't gives information about the types in the array. Just remove it an re-run the tests and see how hard it fails :D therefore we need to specify the type for all arrays.

Comment (slavistan) [reply src/attributes/IQLQuery.ts:23]: The `@serialize()` decorator is empty in a couple of places where array types are used, such as [here](https://github.com/nmshd/cns-content/blob/453b1f376ef08d13cbba711ba33cb57942897d69/src/attributes/types/statement/AbstractIdentityDescriptor.ts#L22). Are theses errors or is there something else I don't understand?

Comment (jkoenig134) [reply src/attributes/IQLQuery.ts:23]: idk

**Pull request merged**